mod_reforged/hooks/skills/backgrounds/character_background.nut

```diff
@@ -53,18 +65,80 @@
+		local baseAttr = this.m.BaseAttributes;
+		local change = this.onChangeAttributes();
+
+		local html = "<div class='attributePredictionHeader'>Base Attribute Ranges for this Background:</div>";
+		html += "<div class='attributePredictionContainer'>";
+		html += formatString("gfx/ui/icons/health.png", baseProperties.Hitpoints, baseAttr.Hitpoints[0] + change.Hitpoints[0], baseAttr.Hitpoints[1] + change.Hitpoints[1]);
+		html += formatString("gfx/ui/icons/melee_skill.png", baseProperties.MeleeSkill, baseAttr.MeleeSkill[0] + change.MeleeSkill[0], baseAttr.MeleeSkill[1] + change.MeleeSkill[1]);
+		html += formatString("gfx/ui/icons/fatigue.png", baseProperties.Stamina, baseAttr.Stamina[0] + change.Stamina[0], baseAttr.Stamina[1] + change.Stamina[1]);
+		html += formatString("gfx/ui/icons/ranged_skill.png", baseProperties.RangedSkill, baseAttr.RangedSkill[0] + change.RangedSkill[0], baseAttr.RangedSkill[1] + change.RangedSkill[1]);
+		html += formatString("gfx/ui/icons/bravery.png", baseProperties.Bravery, baseAttr.Bravery[0] + change.Bravery[0], baseAttr.Bravery[1] + change.Bravery[1]);
+		html += formatString("gfx/ui/icons/melee_defense.png", baseProperties.MeleeDefense, baseAttr.MeleeDefense[0] + change.MeleeDefense[0], baseAttr.MeleeDefense[1] + change.MeleeDefense[1]);
+		html += formatString("gfx/ui/icons/initiative.png", baseProperties.Initiative, baseAttr.Initiative[0] + change.Initiative[0], baseAttr.Initiative[1] + change.Initiative[1]);
+		html += formatString("gfx/ui/icons/ranged_defense.png", baseProperties.RangedDefense, baseAttr.RangedDefense[0] + change.RangedDefense[0], baseAttr.RangedDefense[1] + change.RangedDefense[1]);
+
+		return [{
+			id = 4,
+			type = "description",
+			rawHTMLInText = true,
+			text = html,
+		}];
+	}
 });
 
 ::Reforged.HooksMod.hookTree("scripts/skills/backgrounds/character_background", function(q) {
+	q.getGenericTooltip = @(__original) function()
+	{
+		local ret = __original();
+
+		if (::Reforged.Mod.ModSettings.getSetting("CharacterScreen_ShowBaseAttributeRangesHiring").getValue())
+		{
+			ret.extend(this.getBaseAttributesTooltip(true));
+		}
+
+		return ret;
+	}
+
 	q.getTooltip = @(__original) function()
 	{
 		local ret = __original();
 
-		local player = this.getContainer().getActor();
-		if (::Const.XP.MaxLevelWithPerkpoints - player.getLevel() + player.getLevelUps() > 0)
+		local showBaseAttributeRangeRegular = ::Reforged.Mod.ModSettings.getSetting("CharacterScreen_ShowBaseAttributeRangesRegular").getValue();
+		if (showBaseAttributeRangeRegular == "Always")
+		{
+			ret.extend(this.getBaseAttributesTooltip(false));
+		}
+		else if (showBaseAttributeRangeRegular == "Only New Recruits")
 		{
-			ret.extend(this.getProjectedAttributesTooltip());
+			local player = this.getContainer().getActor();
+			if (player.getLevel() == player.getLevelUps() + 1)	// This condition is an approximation of "not yet having spent any level-up"
```

Comment:
This isn't great; why not just show it whenever they have level ups available?

Reply:
That goes against the advertised "Only New Recruits" option. 
Suddently these ranges would regularly show up on non-new recruits as well.